### ISSUE TITLE
Cannot install System.Collections.Immutable: Install Button does nothing?

### ISSUE THREAD

**Issue by XDracam** [GlitchEnzo/NuGetForUnity#401]
## Description

Originally installed: System.Collections.Immutable 1.7.2 through plugin version 2.0.3 (I think)

Updated package. Clicked "Install All Updates". Worked fine for all packages but that one. For some reason, whenever I click the "Install" button, nothing happens. Of course, the code stopped compiling, as it can't find the immutable collections anymore.

Running the debugger with a breakpoint on any exception reveals this in `NugetPackage.cs`:

![image](https://user-images.githubusercontent.com/30172448/121886309-8e47a780-cd15-11eb-9e08-15032eddac74.png)

- **NuGet Package:** System.Collections.Immutable 5.0.0
- **NuGetForUnity Version:** 3.0.2
- **Unity Version:** 2020.2.2f1
- **Operating System:** Windows 10

**Comment by XDracam** [comment]
Stack trace of exception, copied from debugger:

```
  at System.Enum+EnumResult.SetFailure (System.Enum+ParseFailureKind failure, System.String failureParameter) [0x0001c] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Enum.TryParseEnum (System.Type enumType, System.String value, System.Boolean ignoreCase, System.Enum+EnumResult& parseResult) [0x00059] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Enum.Parse (System.Type enumType, System.String value, System.Boolean ignoreCase) [0x00010] in <9577ac7a62ef43179789031239ba8798>:0 
  at NugetForUnity.NugetPackage.FromNuspec (NugetForUnity.NuspecFile nuspec) [0x0009c] in <ee6cbe4822d046d98553a66d7fbaadcd>:0 
```

**Comment by slimshader** [comment]
Same error here. Enabling verbose logging says Package is already installed and nothing happens.

**Comment by AndreiaSaria** [comment]
The same happened here with System.Threading.Tasks.Extensions.

**Comment by DanielW093** [comment]
I've just come across the same issue - any solutions?

**Comment by XDracam** [comment]
> I've just come across the same issue - any solutions?

The same solution for everything that doesn't work with this plugin: Install the dependency in some other project, manually copy around dlls of the dependency and its dependencies and then hope that unity doesn't complain. Get desperate, manually guess which boxes need to be ticked in the inspector, and at some point it might just work. At least that's how I eventually did it...

**Comment by jhughes2112** [comment]
The simplest answer is to add a text file called `Assets\csc.rsp` and put `-r:System.ServiceModel.dll` in it.  Or whatever assembly you need.

**Comment by bawahakim** [comment]
Same issue on my side. I tried @jhughes2112 solution but doesn't seem to work for me. There is one workaround I found following this:

https://docs.microsoft.com/en-us/visualstudio/gamedev/unity/unity-scripting-upgrade

Basically:
1. Go to the target package: https://www.nuget.org/packages/System.Collections.Immutable/
2. Download package
3. Change extension to `.zip` and extract
4. Find the `dll` file under `lib/netstandard2.0`
5. Extract to `Plugins`
6. **Repeat for every dependency of that package** (yes, incredibly tedious and inefficient)

Works, but ended up not using the Immutables at that point. Could be useful for a critical package.

**Comment by jhughes2112** [comment]
Yes, @bawahakim describes the technique I use now for most dependencies.  I stopped using NuGet because I needed to break apart a package and explicitly manage some dependency version numbers, and NuGet didn't let me do that.  Be aware, though, that if a package requires specific versions of a .dll, Unity will throw a fit if you call a function that is from that package and you have the wrong version (or no version) in the plugins folder, because it fails to load the Assembly, and throws an exception instead.  This won't be apparent until you hit one of those functions at runtime.  So be thorough.

**Comment by IvanMurzak** [comment]
The issue still exists. I like NuGetForUnity. Does it possible to fix it? It is so annoying to import DLLs manually :(

**Comment by yangchos** [comment]
same issue with System.IdentityModel.Tokens.Jwt

**Comment by JoC0de** [comment]
Sorry for the delay. But there was a time with nearly no active contributors...
I tested installing `System.Collections.Immutable` with version 3.1.0 and it works fine, there are many changes from 3.0.5 to 3.1.0 that should have fixed it.
I also tested `System.IdentityModel.Tokens.Jwt` but got some compiler error that the `Microsoft.CSharp` assembly is installed twice. I fixed it in #502 at leas on my machine it worked after the change. @yangchos If you still have problems pleas provide the logs produced wen enabling Verbose logging and the information what ApiCompatibilityLevel you are using (.Net Standard or .Net Framework).

**Comment by yangchos** [comment]
I really appreciate your responce.
It works fine, Thanks!


________________________________
보낸 사람: JoC0de ***@***.***>
보낸 날짜: 2023년 3월 5일 일요일 오전 12:42
받는 사람: GlitchEnzo/NuGetForUnity ***@***.***>
참조: Minjae ***@***.***>; Mention ***@***.***>
제목: Re: [GlitchEnzo/NuGetForUnity] Cannot install System.Collections.Immutable: Install Button does nothing? (#401)


Sorry for the delay. But there was a time with nearly no active contributors...
I tested installing System.Collections.Immutable with version 3.1.0 and it works fine, there are many changes from 3.0.5 to 3.1.0 that should have fixed it.
I also tested System.IdentityModel.Tokens.Jwt but got some compiler error that the Microsoft.CSharp assembly is installed twice. I fixed it in #502<https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGlitchEnzo%2FNuGetForUnity%2Fpull%2F502&data=05%7C01%7C%7C3313a3d052bc4a52ad4b08db1cc7000c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638135413269531683%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WtqhB5nE8JwsRq6jbPPEFYcvHmupLxvaYDvSQ3hRXJI%3D&reserved=0> at leas on my machine it worked after the change. @yangchos<https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fyangchos&data=05%7C01%7C%7C3313a3d052bc4a52ad4b08db1cc7000c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638135413269531683%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Qfdh%2B0QEsW3cjRjW4j1h4fJ%2F0uLT2tQVRKOPHNibVtc%3D&reserved=0> If you still have problems pleas provide the logs produced wen enabling Verbose logging and the information what ApiCompatibilityLevel you are using (.Net Standard or .Net Framework).
